sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+  // Determine what type of incoming twin message this is. Print relevant data for the message.
+  switch (twin_response->response_type)
+  {
+    // A response from a twin GET publish message with the twin document as a payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+      printf("A twin GET response was received\n");
+      if (az_failed(result =
+              parse_twin_desired_temperature_property(twin_payload_span, true, &desired_temp, &version_num)))
+      {
+        // If the item can't be found, the desired temp might not be set so take no action
+        break;
+      }
+      else
+      {
+        send_reported_temperature_property(desired_temp, version_num);
+
+        if (update_device_temp((int32_t)desired_temp))
+        {
+          send_reported_temperature_property(device_max_temp, version_num);
+        }
+      }
+      if (message->payloadlen)
+      {
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+      }
+      break;
+    // An update to the desired properties with the properties as a JSON payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      printf("A twin desired properties message was received\n");
+      printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+
+      // Get the new temperature
+      if (az_failed(
+              result = parse_twin_desired_temperature_property(
+                  twin_payload_span, false, &desired_temp, &version_num)))
+      {
+        printf("Could not parse desired temperature property, az_result %04x\n", result);
+        break;
+      }
+      send_reported_temperature_property(desired_temp, version_num);
+
+      if (update_device_temp((int32_t)desired_temp))
+      {
+        send_reported_temperature_property(device_max_temp, -1);
+      }
+      break;
+
+    // A response from a twin reported properties publish message. With a successfull update of
+    // the reported properties, the payload will be empty and the status will be 204.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties response message was received\n");
+      break;
+  }
+}
+
+// Invoke the requested command if supported and return status | Return error otherwise
+static void handle_command_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_method_request* command_request)
+{
+
+  if (az_span_is_content_equal(report_command_name_span, command_request->name))
+  {
+    az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+    az_span command_payload_span
+        = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+    // Invoke command
+    uint16_t return_code;
+    az_result response = invoke_getMaxMinReport(
+        command_payload_span, command_response_span, &command_response_span);
+    if (response == AZ_ERROR_ITEM_NOT_FOUND)
```

Comment:
>response [](start = 8, length = 8)

If invoke_getMaxMinReport ends up returning AZ_ERROR_PARSER_UNEXPECTED_CHAR, which it theoretically could, we'll end up setting response_code=200.

You might conside making invoke_getMinMaxReport() return the http status error code itself since it's going to know that the AZ_ERROR_PARSER_UNEXPECTED_CHAR signifies a 400 presumably whereas if we ran out of memory (OK I get that's not how this SDK works...) that's a 500.  Or alternately if we know we'll always either be a 200 or a 400 at this stage, you could haev a result=OK => 200.  I think we're past the point of a 404 because the check in is_content_eqal at top of this block to make sure we're report_command_name_span.

Reply:
I think at this point making it a `400` probably is sufficient. Here I'll make it
```c
if (response != AZ_OK)
    return_code = 400;
```